### PR TITLE
Added more options, changed hashmap, cleaned up code

### PR DESCRIPTION
**Pull request by enerko**
I added:
- A "reset" option after game is over
- A "go back to title screen" option in game over and level clear screens

I changed:
- The structure of the controller -> maze interactor so that the controller is just responsible for passing in the input to the use case layer, and it does not have to know anything about the implementation in the use cases
- The class name for Maze to MazeInfo
- The hashmap in MazeInfo to a list of constants and getter methods
- Fixed stamina bug where the stamina continues to decrease when the player is walking into an obstacle
- The appearance of the screens, still in progress

Currently there is a bug where the enemy speeds up after every game which is happening because I moved the startGameThread call to the load method. I think it makes sense that the game thread starts after loading the game, just to make sure there are no bugs before the user actually starts the game. @ltenenb do you have any ideas how to fix it?

### CODE REVIEW

**Review by tseamuscorlett — APPROVED** [review]
Awesome! Just left a few questions/comments about the minor details

File: src/test/java/items/MazeInfoItemsTest.java [hunk 1]
```diff
@@ -8,7 +8,7 @@
 import use_cases.items.MazeItems;
 
 /** Test the MazeItems class */
-public class MazeItemsTest {
+public class MazeInfoItemsTest {
```
[the remark this answers is not in the document]
Comment (enerko) [reply src/test/java/items/MazeInfoItemsTest.java:11]: Sorry that was unintentional, I think it automatically changed it when I renamed the Maze class. I will change it.

File: src/main/java/use_cases/default_game/MazeInteractor.java [hunk 2]
```diff
@@ -210,39 +242,43 @@ private synchronized void updateHazards() {
...
-        }
-        else if(keycode == KeyEvent.VK_2){
+        } else if (keycode == KeyEvent.VK_2) {
             load("mazes/MediumMaze.txt");
             mazeLevel = "MEDIUM";
-        }
-        else if(keycode == KeyEvent.VK_3){
+        } else if (keycode == KeyEvent.VK_3) {
             load("mazes/HardMaze.txt");
             mazeLevel = "HARD";
```
[the remark this answers is not in the document]
Comment (enerko) [reply src/main/java/use_cases/default_game/MazeInteractor.java:283]: Yes the leaderboard would need to access the levels somehow. The leaderboard part is not implemented yet so it’s not being accessed right now. There might be a better way to store the levels though, if you have any ideas let me know!

File: src/main/java/use_cases/default_game/MazeInteractor.java [hunk 3]
```diff
@@ -210,39 +242,43 @@ private synchronized void updateHazards() {
...
+            outputBoundary.changeState(IGamePanelOutputBoundary.GAME_OVER_STATE);
         }
     }
 
-    /** Is the game over? */
+    /**
+     * Is the game over?
+     */
     public synchronized boolean gameOver() {
         return isPlayerKilled() || player.getStageClear();
```
Comment (tseamuscorlett) [inline src/main/java/use_cases/default_game/MazeInteractor.java:259]: we need another condition || player.getStamina <= 0

File: src/main/java/use_cases/default_game/MazeInteractor.java [hunk 4]
```diff
@@ -210,39 +242,43 @@ private synchronized void updateHazards() {
         checkPlayerKilled();
     }
 
-    /** Check if the player has been killed by an enemy. */
-    private synchronized void checkPlayerKilled() {
+    /**
+     * Check if the player has been killed by an enemy.
```
Comment (tseamuscorlett) [inline src/main/java/use_cases/default_game/MazeInteractor.java:246]: "or has run out of Stamina"

**Review by tseamuscorlett — APPROVED** [review]

**Review by tseamuscorlett — APPROVED** [review]

**Pull request merged**